initramfs/src/registry.rs

```diff
@@ -50,48 +154,180 @@ impl Registry {
+        })?;
 
-        debug!("Successfully got layers : {:?}", layers_metadata);
+        debug!("Manifest type : {:?}", manifest_type);
+
+        let manifest = match manifest_type {
+            ManifestType::ManifestV2 => res
+                .json::<ManifestV2Response>()
+                .await
+                .map_err(|e| anyhow!(e).context("Failed to parse response"))?,
+            ManifestType::ManifestListV2 => {
+                let manifest = res
+                    .json::<ManifestListV2Response>()
+                    .await
+                    .map_err(|e| anyhow!(e).context("Failed to parse response"))?;
+
+                self.manifest_from_manifest_v2_list(manifest, image_name)
+                    .await?
+            }
+            ManifestType::OCIv1List => {
+                let manifest = res
+                    .json::<OCIv1ListResponse>()
+                    .await
+                    .map_err(|e| anyhow!(e).context("Failed to parse response"))?;
+
+                self.manifest_from_oci_list(manifest, image_name).await?
+            }
+            ManifestType::OCIv1 => unimplemented!(),
+        };
+
+        let layers_metadata = manifest
+            .layers
+            .into_iter()
+            .map(|layer| LayerMetadata {
+                blob_sum: layer.digest.clone(),
+            })
+            .collect::<Vec<_>>();
+
+        debug!(
+            "Successfully got layers : \n\t{}",
+            layers_metadata
+                .iter()
+                .map(|layer| layer.blob_sum.clone())
+                .collect::<Vec<String>>()
+                .join("\n\t")
+        );
 
         Ok(layers_metadata)
     }
 
-    async fn get_layer(
+    async fn manifest_from_manifest_v2_list(
+        &self,
+        manifest: ManifestListV2Response,
+        image_name: &str,
+    ) -> Result<ManifestV2Response> {
+        let sub_manifest = manifest
+            .manifests
+            .iter()
+            .find(|manifest| {
+                manifest
+                    .platform
+                    .as_ref()
+                    .map(|platform| platform.os == "linux" && platform.architecture == "amd64")
+                    .unwrap_or(false)
+            })
+            .map(|manifest| manifest.digest.clone())
+            .ok_or(anyhow!("No manifest found"))?;
+
+        trace!("Sub manifest : {:?}", sub_manifest);
+
+        self.manifest_from_oci(sub_manifest, image_name).await
+    }
+
+    async fn manifest_from_oci_list(
+        &self,
+        oci_manifest: OCIv1ListResponse,
+        image_name: &str,
+    ) -> Result<ManifestV2Response> {
+        let sub_manifest = oci_manifest
+            .manifests
+            .into_iter()
+            .find(|manifest| {
+                manifest
+                    .platform
+                    .as_ref()
+                    .map(|platform| platform.os == "linux" && platform.architecture == "amd64")
```

Comment:
maybe it would be good to have "linux" and "amd64" as const since we use them in multiple places

Reply:
Honestly, both ways are bad IMO. The correct way of doing it would be to be able to specify the arch/os the user wants as params to the program. This is out of the scope of this PR. Should I create a const in the meantime in your opinion ?

Reply:
IMO, yes, it would be a clean way to wait for something better ;)

Reply:
Will do.